dali/base/dautils.cpp

```diff
@@ -119,7 +119,25 @@ IPropertyTree * getDropZonePlane(const char * name)
 
 bool isPathInPlane(IPropertyTree *plane, const char *path)
 {
-    return isEmptyString(path) || startsWith(path, plane->queryProp("@prefix"));
+    if (isEmptyString(path))
+        return true;
+
+    const char *prefix = plane->queryProp("@prefix");
+    while (*prefix && *prefix == *path)
+    {
+        path++;
+        prefix++;
+    }
+    if (0 == *prefix)
+    {
+        if (0 == *path || isPathSepChar(*path))
+            return true;
+        if (isPathSepChar(*(path - 1))) //implies both last characters of prefix and path were '/'
```

Comment:
There's a theoretical problem here, if prefix is an empty string and path isn't, then it will reach here and dereference invalid memory.
Thi can only happen if prefix is empty, so I'd probably guard against by checking if prefix is empty at the start:
```
if (isEmptyString(prefix))
    return false; // prefix is empty, path is not - can't match.
```

Reply:
I will add those 2 lines. It is a 'theoretical problem'. If the prefix is empty by some mistake, we should return the false. Otherwise, it will be a big security hole if all paths are allowed.